sql/core/src/main/scala/org/apache/spark/sql/UpdateWriter.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.annotation.Experimental
+import org.apache.spark.sql.catalyst.plans.logical.{Assignment, UpdateTable}
+import org.apache.spark.sql.functions.expr
+
+/**
+ * `UpdateWriter` provides methods to define and execute an update action on a target table.
+ *
+ * @param tableDF DataFrame representing table to update.
+ *
+ * @since 4.0.0
+ */
+@Experimental
+case class UpdateWriter (tableDF: DataFrame) {
```

Comment:
let's avoid using case class as the API, `MergeIntoWriter` is also a normal class